### PR TITLE
chore: refactor intercom proxy tests to new component structure

### PR DESCRIPTION
**Pull request by mihir-4116**
## What are the changes introduced in this PR?

Write a brief explainer on your code changes.

## What is the related Linear task?

Resolves INT-XXX

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3088/test-report.html

File: test/integrations/destinations/intercom/dataDelivery/business.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        status: 200,
+        body: {
+          output: {
+            message:
+              '[Intercom Response Handler] Request failed for destination intercom with status: 408',
+            response: [
+              {
+                error:
+                  '{"type":"error.list","request_id":"000on04msi4jpk7d3u60","errors":[{"code":"Request Timeout","message":"The server would not wait any longer for the client"}]}',
+                metadata: {
```
Comment (koladilip) [inline test/integrations/destinations/intercom/dataDelivery/business.ts:142]: can we use proxyMetdata

File: test/integrations/destinations/intercom/dataDelivery/business.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                  {
+                    code: 'Request Timeout',
+                    message: 'The server would not wait any longer for the client',
+                  },
+                ],
+              },
+              status: 408,
+            },
+            statTags: {
+              destType: 'INTERCOM',
```
Comment (koladilip) [inline test/integrations/destinations/intercom/dataDelivery/business.ts:93]: can we define a reusable function for this to generate this

File: test/integrations/destinations/intercom/dataDelivery/business.ts (hunk 1 above)
Comment (mihir-4116) [reply test/integrations/destinations/intercom/dataDelivery/business.ts]: addressed

File: test/integrations/destinations/intercom/dataDelivery/business.ts (hunk 2 above)
Comment (mihir-4116) [reply test/integrations/destinations/intercom/dataDelivery/business.ts]: addressed

**Review by koladilip — APPROVED** [review]

File: test/integrations/destinations/intercom/dataDelivery/business.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  workspaceId: 'default-workspaceId',
+};
+
+const metadataArray = [generateMetadata(1)];
+
+export const testScenariosForV0API = [
+  {
+    id: 'intercom_v0_other_scenario_1',
+    name: 'intercom',
+    description: '[Proxy v0 API] :: Scenario for testing Request Timeout error from destination',
```
Comment (utsabc) [inline test/integrations/destinations/intercom/dataDelivery/business.ts:59]: These scenarios should be add as `other.ts` file separtely

**Review by Gauravudia — DISMISSED** [review]

**Comment by utsabc** [comment]
I see only a retry scenario is just added we have to test the `application` level scenarios for intercom, I see some refs here
https://developers.intercom.com/docs/references/rest-api/errors/http-responses/
https://developers.intercom.com/docs/references/rest-api/errors/error-objects/
https://developers.intercom.com/docs/references/rest-api/errors/error-codes/

can we mock at-least some of these responses which we see important during delivery and need to be handled should be tested.

For network errors like 5xx these should be added in `other.ts` file. Please check a reference integration like campaign manager/ braze

File: test/integrations/destinations/intercom/dataDelivery/business.ts (hunk 3 above)
Comment (mihir-4116) [reply test/integrations/destinations/intercom/dataDelivery/business.ts]: addressed

File: test/integrations/destinations/intercom/network.ts [hunk 4]
```diff
@@ -425,5 +416,197 @@ const deliveryCallsData = [
...
+        update_last_request_at: true,
+        user_id: 'test_user_id_1',
+        custom_attributes: {
+          'address.city': 'Kolkata',
+          'address.state': 'West Bengal',
+        },
+      },
+      params: {},
+      headers: {
+        'Content-Type': 'application/json',
```
Comment (koladilip) [inline test/integrations/destinations/intercom/network.ts:590]: this is repeated

File: test/integrations/destinations/intercom/network.ts [hunk 5]
```diff
@@ -425,5 +416,197 @@ const deliveryCallsData = [
...
+        ],
+      },
+      status: 409,
+    },
+  },
+  {
+    httpReq: {
+      url: 'https://api.intercom.io/users',
+      data: {
+        email: 'test_1@test.com',
```
Comment (koladilip) [inline test/integrations/destinations/intercom/network.ts:576]: some data looks repeated

File: test/integrations/destinations/intercom/network.ts (hunk 4 above)
Comment (mihir-4116) [reply test/integrations/destinations/intercom/network.ts]: addressed

File: test/integrations/destinations/intercom/network.ts (hunk 5 above)
Comment (mihir-4116) [reply test/integrations/destinations/intercom/network.ts]: addressed

**Review by koladilip — DISMISSED** [review]

File: test/integrations/destinations/intercom/dataDelivery/other.ts [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+            },
+            statTags: expectedStatTags,
+          },
+        },
+      },
+    },
+  },
+];
+
+export const otherScenariosV1: ProxyV1TestData[] = [
```
Comment (utsabc) [inline test/integrations/destinations/intercom/dataDelivery/other.ts:95]: You should add the network scenarios which are applicable check `other.ts` of `braze` as reference

Comment (mihir-4116) [reply test/integrations/destinations/intercom/dataDelivery/other.ts:216]: addressed

File: test/integrations/destinations/intercom/dataDelivery/business.ts [hunk 7]
```diff
@@ -0,0 +1,615 @@
...
+        ),
+        method: 'POST',
+      },
+    },
+    output: {
+      response: {
+        status: 200,
+        body: {
+          output: {
+            message: 'Request Processed Successfully',
```
[the remark this answers is not in the document]
Comment (mihir-4116) [reply test/integrations/destinations/intercom/dataDelivery/business.ts:385]: added it as backlog item

File: test/integrations/destinations/intercom/dataDelivery/business.ts [hunk 8]
```diff
@@ -0,0 +1,615 @@
...
```
[the remark this answers is not in the document]
Comment (mihir-4116) [reply test/integrations/destinations/intercom/dataDelivery/business.ts:427]: added it as backlog item

**Review by utsabc — APPROVED** [review]

**Review by Gauravudia — APPROVED** [review]

**Pull request merged**